Revamping the Root Level README

dwahme opened the pull request:
This addresses #402 

This adds basic information about chiventure to the root level readme, such as how to build and run chiventure, as well as an overview of chiventure and its various components. It also gives info on where to start with making a WDL file

ecoble commented on docs/README.md:
```diff
@@ -1,4 +1,76 @@
+
+#### Action Management
+
+During the game, a player will be able to perform actions (pushing, pulling, opening, closing, etc.) that will usually affect the state of the game in some way. This component is responsible for processing actions like “take the orb” (and checking whether it is possible to perform that action)
```
nit: currently in the game this command has to be "take orb"

ecoble commented on docs/README.md:
````diff
@@ -1,4 +1,76 @@
+./chiventure
+```
+
+Once inside chiventure, you can type `HELP` for the list of available commands. In the future, you can load a game from inside chiventure with `LOAD <path_to_wdl_file>`
````
There are two different commands, "load" and "load_wdl" (confusing and ideally I think they should be one command) but right now "load_wdl" is the one that can load wdl files from within Chiventure in the way described here.

dwahme replied:
Oh, totally missed it since I was just reading off the help screen. As an FYI, currently getting
```
 > LOAD_WDL ../tests/wdl/examples/connected_rooms.yaml                                                                                                        
chiventure: /home/dawson/dev/chiventure/src/libobj/src/parser.c:158: parse_game: Assertion `pfile' failed.                                                     
                                                                                                          fish: “./chiventure” terminated by signal SIGABRT (Abort)
```

ecoble replied:
Hmm I am not having this issue when I try it on master

ecoble commented on docs/README.md:
````diff
@@ -1,4 +1,76 @@
+
+To try a sample game, from inside the build directory, run 
+```
+./chiventure ../tests/wdl/examples/connected_rooms.yaml
````
Should we rename the .yaml files to be .wdl for consistency? @borjasotomayor

borjasotomayor replied:
Yes, I think this would make sense.

dwahme commented:
Most changes addressed, waiting for @borjasotomayor's opinion on connected_rooms.yaml being renamed to `connected_rooms.wdl` (and similar for `test.yaml`).

Also, should that be included in this PR if it needs to be changed?

borjasotomayor commented on docs/README.md:
```diff
@@ -1,4 +1,79 @@
+
+Example games can be found in `tests/wdl/examples/`, such as [test.yaml](/tests/wdl/examples/test.yaml)
+
+## (For Developers) Component Overview
```
This whole section should be refactored into a "Developer Guide" document (distinct from the developer guidelines, which will probably just end up being moved into the developer guide)

dwahme replied:
Moved the Component Overview section to /docs/developer.md since it seems like we wanted to merge the two anyways

borjasotomayor commented on docs/README.md:
```diff
@@ -1,4 +1,79 @@
+
+This component is responsible for any non-CLI elements of the user interface, such as visualizing a map of the game or displaying graphics associated with individual rooms
+
+## Miscellaneous
```
Rename this to "Developer Resources" and include additional links to pertinent resources in `docs/`

dwahme replied:
Renamed, added link to `/docs/README.md` which can serve as the index file for `/docs/`

dwahme commented:
Addressing Borja's requests
- I've moved the root level README to `/README.md`
- Moved the component overview section from `/README.md` to `/docs/developer.md`
- Created `/docs/README.md` to serve as an index page for the `/docs/` directory
- Renamed `*.yaml` files to `*.wdl`

ecoble commented on README.md:
````diff
@@ -0,0 +1,52 @@
+./chiventure
+```
+
+However, at the moment, it is not possible to load a WDL file from inside chiventure.
````
The load_wdl command on master works for me, but I am using last year's UChicago VM (not sure if that would make a difference at all)

borjasotomayor replied:
I'm getting the following error when doing this on master:

```
  > load_wdl ../tests/wdl/examples/test.yaml
chiventure: /home/borja/eclipse_workspace/chiventure/src/libobj/src/parser.c:158: parse_game: Assertion `pfile' failed.
                                                                                                                       Aborted (core dumped)
```

This is also the same error you get if you specify a non-existent file:

```
  > load_wdl foo
chiventure: /home/borja/eclipse_workspace/chiventure/src/libobj/src/parser.c:158: parse_game: Assertion `pfile' failed.
                                                                                                                       Aborted (core dumped)
```

I'm going to create a separate issue for this; we don't need to hold up the README on account of this.

borjasotomayor replied:
The issue to keep track of the load_wdl issue is #411
